### PR TITLE
[query] add StreamTake node

### PR DESCRIPTION
**Pull request by patrick-schultz**
~~stacked on #8519~~

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/EmitStream.scala [hunk 1]
```diff
@@ -193,27 +193,43 @@ object CodeStream { self =>
       CodeStream.flatMap(map(f))
   }
 
-  def range(mb: EmitMethodBuilder[_], start: Code[Int], step: Code[Int], len: Code[Int]): Stream[Code[Int]] = {
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/expr/ir/EmitStream.scala:194]: I didn't end up using iota for StreamTake, but I anticipate it being useful for other nodes, so I left it in.

**Comment by tpoterba** [comment]
needs rebase, though

File: hail/src/main/scala/is/hail/expr/ir/EmitStream.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/EmitStream.scala:196]: what does the name come from? This is just an infinite range stream, right?

Comment (patrick-schultz) [reply hail/src/main/scala/is/hail/expr/ir/EmitStream.scala:196]: It's a traditional name in programming languages for the infinite range function. It started in APL, and is used in C++ and others. Happy to give it a more descriptive name.

**Comment by patrick-schultz** [comment]
There is a PruneDeadFields rule. The default case for rebuild should work, right?

I did forget the PruneDeadFields test. I'll add that to the checklist too.

**Comment by tpoterba** [comment]
oops, there totally is a prune rule. Default rebuild will be fine, yeah.

**Comment by patrick-schultz** [comment]
Added Pruner tests, and update the add-ir-checklist.

**Comment by johnc1231** [comment]
Failing Java tests (also, looking at the `test_hail_java` log, looks like we print a lot during this stream stuff?)

**Comment by patrick-schultz** [comment]
Just fixed, I forgot the parser rule. Yeah, I left in some printing in the stream stuff that I should probably at least flag to turn off in automated testing.

**Pull request merged**